### ISSUE TITLE
Intel HD Graphics for Intel Celeron N3050 not working

### ISSUE THREAD

**Issue by redkill3r42604** [arnoldthebat/chromiumos#65]
Blank screen when booting Chromium from USB. Ctrl+Alt+F2 brings up console, which does show up.

**Comment by bscoggins** [comment]
This issue does not occur in the May 21 2016 daily builds but does occur in the May 21 2016 special builds.

When booting the chromium splash screen displays and is animated.  After this is displayed for a bit the screen dumps back to a text screen ending with "Booting the kernel."  Ctrl+Alt+F2 works for console.

dmesg shows the following messages regarding gpu:
[drm] stuck on render ring
[drm] GPU HANG: ecode 8:0:0xab2ffff9, in X [1472], reason: Ring hung, action: reset
drm/i915: Resetting chip after gpu hang

Found similar issues and forms of resolution (https://bbs.archlinux.org/viewtopic.php?id=203123) and (https://mtekk.us/archives/guides/working-around-drm-stuck-on-render-ring-on-intel-gma/).  Neither worked in this case.

Booting from USB on an Acer Cloudbook 14 (2GB ram, N3050 processor).

**Comment by redkill3r42604** [comment]
The laptop I am using is an Acer Aspire R3 (specifically the R3-131T-C1YF) which has the same specifications as the Acer Cloudbook 14 I believe.

**Comment by redkill3r42604** [comment]
Does that mean I should use the daily builds or will I still run into the same problem?

**Comment by bscoggins** [comment]
With the Cloudbook 14, the daily build starts just fine and the display works great, but wireless is not supported (Intel AC 3160).  Wireless was added to the special build recently but the display does not work on that one.  For the Aspire R3 it looks like it uses the Intel AC 3165, no idea if that works on the daily build or not but it seems like the Intel graphics should work.

**Comment by mikeandmore** [comment]
I have exactly the same problem. I tried ``intel_iommu=soft`` and ``intel_iommu=igfx_off``. None of them work.

**Comment by redkill3r42604** [comment]
So, after 2 days of testing and research, I think I'll be able to find a solution soon. Here's what happened:

My laptop, as I mentioned, is the Acer Aspire R3-131T-C1YF. It has an Intel Celeron N3050, 2 GB RAM, Intel HD Graphics, and an eMMC (Hynix HGB4e). It originally came with Windows 10. Since I want to switch to Chromium OS on this laptop, I searched for builds that are active. CloudReady and Arnoldthebat's Chromium OS are two that I have found. I have first tried Arnoldthebat's Chromium OS Special Builds (x64), but booting it lead me to a black screen (Ctrl+Alt+F2 shows console). I haven't tried the x86 builds, but I think those will also lead me to a black screen. I then tried CloudReady, but that also lead me to a black screen! This time, Ctrl+Alt+F2 didn't do anything. After this ripping-my-hair moment, I tried Arnoldthebat's Chromium OS Daily Builds (x64). When I booted up the USB, the display actually showed the Chromium OS start screen! The only thing that didn't work was WiFi, I use an Intel Wireless AC 3165. After 5 hours, I still couldn't get WiFi working! I was getting really mad at this point, so I went and slept. The next day, I decided to try the x86 version of Arnoldthebat's Chromium OS Daily Builds. This time, when I booted it, the display showed the Chromium OS start screen AND the WiFi works (it would disable itself after a few seconds, but it would enable if you plugged in an active Ethernet and unplugged it. This was because of APManager, which is easy to disable)! But the only problem was (and this was when I started to really get mad) that Chromium OS wouldn't find my eMMC! But when I use GParted Live, it sees it perfectly! What I am going to do next is to make a new Chromium OS USB and clone it to the eMMC (using Clonezilla) and see if it works. Wish me luck and I will post an update soon if it works! (...I hope this works)

**Comment by redkill3r42604** [comment]
UPDATE: Looks like it doesn't work. :( When booting from the eMMC (hard drive), it gets stuck at "Booting 'local image A'". Guess I'll use Cub Linux on my Acer R3 for now.

**Comment by pzman89** [comment]
Just wanted to add I have the same Acer a01-431 with Intel Celeron N-3050 processor and wifi as mentioned in this post (cloudready 14" Windows 10 machine) and am running into the same issue as others posted.

**Comment by redkill3r42604** [comment]
Hopefully Arnold will be able to fix this soon as it looks like quite a few people have this problem!

**Comment by mikeandmore** [comment]
I had to boot with ``no_timer_check`` for all the time. If your kernel stuck at boot, you should try this.

**Comment by arnoldthebat** [comment]
@mikeandmore - did that fix the graphics?

**Comment by mikeandmore** [comment]
@arnoldthebat No, that didn't fix the graphics. It only allow the kernel to boot. Later, it will show the GPU hang message in dmesg.

**Comment by mikeandmore** [comment]
I guess we could start from the daily release since the graphics works fine there.

Is there any way we can compile the kernel modules (or the kernel images) ourselves? I could install Linux flawlessly on these computers. I'm think this is just a kernel version/config issue.

**Comment by arnoldthebat** [comment]
Hi Mike

The daily builds use the Freon drivers for graphics. Its somehting im
pondering on releasing as a fork from the specials. Bear with me whilst I
sort it.

**Comment by arnoldthebat** [comment]
The latest vanilla builds referenced in http://arnoldthebat.co.uk/wordpress/2016/08/07/chromiumos-special-build-r54-8659/ should give you working graphics plus google drive integration

**Comment by redkill3r42604** [comment]
Will test soon. But out of curiosity, does the vanilla build support the Intel Wireless AC 3165? Since on the post of the vanilla build, it states "WiFi and Ethernet will also have reduced support."

**Comment by mikeandmore** [comment]
@redkill3r42604 @arnoldthebat I successfully installed r54-8659 today.

With this special build (freon), I still cannot use the wifi out of box. It seems that ``iwlwifi.ko`` is in the build, but the firmware for Intel 3160 is missing. ``iwlwifi.ko`` insists on loading ``iwlwifi-3160-17.ucode``, but the build only contains ``iwlwifi-3160-16.ucode``. So I grabbed one from https://github.com/OpenELEC/iwlwifi-firmware/tree/master/firmware , and then the wifi works flawlessly!

Still, I need to boot with ``no_timer_check`` all the time, but that's simple. It's just adding an option to the kernel.
